### PR TITLE
[fa] Recurrent neural networks

### PR DESCRIPTION
**Pull request by AlisterTA**
The first draft is done. @erfannoury please review it and let me know if there are any changes you would like me to make.

### CODE REVIEW

**Comment by erfannoury** [comment]
I will review it at the latest until the end of next week.

File: fa/recurrent-neural-networks.md [hunk 1]
```diff
@@ -0,0 +1,868 @@
...
+[عیب جستجوی پرتو، عیب  RNN، افزایش پهنای پرتو، امتحان معماری‌های مختلف، تنظیم، جمع‌آوری داده‌های بیشتر]
+</div>
+
+<br>
+
+
+**81. Bleu score ― The bilingual evaluation understudy (bleu) score quantifies how good a machine translation is by computing a similarity score based on n-gram precision. It is defined as follows:**
+
+<div dir="rtl">
+امتیاز Bleu ― جایگزین ارزشیابی دوزبانه  (bleu) میزان خوب بودن ترجمه ماشینی را با محاسبه‌ی امتیاز تشابه برمبنای دقت ان گرام اندازه‌گیری می‌کند. بصورت زیر تعریف میشود:
```
Comment (erfannoury) [inline fa/recurrent-neural-networks.md]: ان‌گرام
(این امتیاز)‌به صورت زیر تعریف می‌شود

File: fa/recurrent-neural-networks.md [hunk 2]
```diff
@@ -0,0 +1,868 @@
...
+که pn امتیاز bleu تنها براساس ان گرام است و بصورت زیر تعریف می‌شود:
+</div>
+
+<br>
+
+
+**83. Remark: a brevity penalty may be applied to short predicted translations to prevent an artificially inflated bleu score.**
+
+<div dir="rtl">
+تذکر: ممکن است برای پیشگیری از امتیاز اغراق آمیز تصنعیbleu ، برای ترجمه‌های پیش‌بینی‌شده‌ی کوتاه از جریمه اجمالی (مختصر) استفاده شود.
```
Comment (erfannoury) [inline fa/recurrent-neural-networks.md]: از جریمه‌ی اختصار استفاده شود

File: fa/recurrent-neural-networks.md [hunk 3]
```diff
@@ -0,0 +1,868 @@
...
+</div>
+
+<br>
+
+
+**84. Attention**
+
+<div dir="rtl">
+توجه
```
Comment (erfannoury) [inline fa/recurrent-neural-networks.md]: Maybe ژرف‌بینی

File: fa/recurrent-neural-networks.md [hunk 4]
```diff
@@ -0,0 +1,868 @@
...
+با
+</div>
+
+<br>
+
+
+**87. Remark: the attention scores are commonly used in image captioning and machine translation.**
+
+<div dir="rtl">
+نکته: امتیازات توجه عموما در برچسب‌گذاری تصویر (image captioning) و ترجمه ماشینی کاربرد دارد.
```
Comment (erfannoury) [inline fa/recurrent-neural-networks.md]: Image captioning -> عنوان‌سازی متنی برای تصویر

File: fa/recurrent-neural-networks.md [hunk 5]
```diff
@@ -0,0 +1,868 @@
...
+وزن توجه ― مقدار توجهی که خروجی y باید به فعال‌سازی a<t′> داشته باشد به‌وسیله‌ی α<t,t′> به‌دست می‌آید که به‌صورت زیر محاسبه می‌شود:
+</div>
+
+<br>
+
+
+**90. Remark: computation complexity is quadratic with respect to Tx.**
+
+<div dir="rtl">
+نکته: پیچیدگی محاسبه به نسبت Tx از نوع درجه‌ی دوم است.
```
Comment (erfannoury) [inline fa/recurrent-neural-networks.md]: پیچیدگی محاسباتی

**Comment by AlisterTA** [comment]
@erfannoury do you think `ژرف‌بینی`  defines the meaning of the word ` Attention`  correctly? how about `ژرف نگری` ?

Considering the words `encoder` and `decoder` ,if you know of better alternatives for our use-case please let me know.

@shervinea could you have a look at the following paragraph, it doesn't seem to be correct? according to @erfannoury <i>"Actually LSTM is not a generalization of GRU, GRU is a specialization of LSTM (LSTM precedes GRU, but the original sentence implies it's the reverse)."</i>

> 36. GRU/LSTM ― Gated Recurrent Unit (GRU) and Long Short-Term Memory units (LSTM) deal with the vanishing gradient problem encountered by traditional RNNs, <b> with LSTM being a generalization of GRU</b> . Below is a table summing up the characterizing equations of each architecture:

it should change to :

> 36. GRU/LSTM ― Gated Recurrent Unit (GRU) and Long Short-Term Memory units (LSTM) deal with the vanishing gradient problem encountered by traditional RNNs, <b> with GRU being a generalization of LSTM</b> . Below is a table summing up the characterizing equations of each architecture:


and i'm not sure what `Noted ow` means in the following picture:

untitled

**Comment by erfannoury** [comment]
@AlisterTA yes I think `ژرف‌نگری` is a better translation. Let's use that then.
Also I agree with the translation for "Noted ow."

**Comment by erfannoury** [comment]
@AlisterTA thank you again. I checked the translation once more and it's ready to be merged (after the two remaining issues are resolved with @shervinea).

**Comment by shervinea** [comment]
Thank you @erfannoury and @AlisterTA for your awesome work. Regarding discussion about content:

1. The sentence "LSTM is a generalization of GRU" underlines the fact that LSTM is more general than GRU (cf. equations characterizing these two types of networks).

2. Thanks for the catch, this term should indeed be `Named-entity recognition`.

3. o_w denotes the word vector w in one-hot representation, and e_w the word vector w in a word embedding representation.

**Comment by AlisterTA** [comment]
@shervinea Done!

**Comment by erfannoury** [comment]
Thank you @AlisterTA 👍

**Comment by shervinea** [comment]
Thank you again for your awesome work @AlisterTA and @erfannoury.

**Pull request closed without merging**